tests/python/multidevice/linear.py

```diff
@@ -61,11 +100,47 @@ def forward(
     @staticmethod
     def backward(ctx, grad_output: DTensor):
         input, weight = ctx.saved_tensors
+        assert input.dim() in [2, 3]
 
-        op = FusionDefinitionWrapper(
-            partial(
-                define_linear_backward, LinearConfig(weight.size(1), weight.size(0))
-            )
+        op = get_fusion_definition_wrapper(
+            ComputeType.BACKWARD,
+            LinearConfig(weight.size(1), weight.size(0), input.dim() == 3),
         )
         grad_x, grad_w = op([input, weight, grad_output])
         return (grad_x, grad_w)
+
+
+class TensorParallelLinear(torch.nn.Linear):
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int,
+        in_placements: Iterable[Placement] = [],
+    ):
+        super().__init__(in_features, out_features, bias=False)
+        self.in_placements = in_placements
+
+    def __repr__(self):
+        base_repr = super().__repr__()
+        return f"{base_repr[:-1]}, in_placements={self.in_placements}, weight_placements={self.weight.data.placements})"
+
+    @classmethod
+    def distribute(
+        cls,
+        linear: torch.nn.Linear,
+        mesh: dist.device_mesh.DeviceMesh,
+        in_placements: Iterable[Placement] = [],
+        weight_placements: Iterable[Placement] = [],
+    ):
+        tp_linear = cls(linear.in_features, linear.out_features, in_placements)
+        tp_linear.weight = torch.nn.Parameter(
```

Comment:
`weights` will be a fusion input and hence, given to us.
So should we expect weights to be provided in the initialization of this method, and check their shardings in the `parallelize_linear_with_nvfuser` module?